### PR TITLE
add views

### PR DESCRIPTION
**Pull request by BANCUNGUYE66**
### Implement views

- [x] Implement the design from the sneak peek wireframes.
- [x] Use methods that I have created in the [processing data in models].
- [x]  Add some data to database.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1 +1,15 @@
...
+  <p class='no-posts'>No posts</p>
+<% else %>
+<% @posts.each do |post|%>
+<li>
+  <%= render 'partials/post', post: post %>
+  <%= render 'partials/comment', comments: post.recent_comments %>
+</li>
+<%end%>
+<%end%>
+</ul>
```
Comment (BenMukebo) [inline app/views/posts/index.html.erb]: - [ ] Kindy fix this error that is happening on the post index view, I would strongly suggest you revise the association on your models to solve this

image

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,22 @@
...
+
+Sed in elementum libero. Maecenas eleifend justo eget mauris ultricies, in posuere orci bibendum. Cras vel justo metus. Vivamus non fringilla erat. Nullam auctor sit amet neque sit amet auctor. Vivamus eget nisl a massa viverra convallis ac quis libero. Curabitur hendrerit, velit eget efficitur fermentum, mi eros interdum nisl, ut lacinia urna ipsum in lectus. Integer sollicitudin, lectus eget convallis finibus, elit risus blandit tellus, eget euismod turpis velit ac nibh. Integer scelerisque dui nec justo auctor, et egestas est scelerisque. Aenean in dolor at arcu venenatis finibus.
+
+Sed tincidunt, arcu vel sollicitudin semper, purus nunc bibendum neque, vel consectetur nisi elit et libero. Sed mollis egestas turpis, eu varius orci fermentum eget. Aliquam quis arcu at nisi venenatis feugiat. Sed in lectus ac lectus scelerisque malesuada. Vivamus semper')
+
+6.times do
+    Comment.create(user_id: user_one.id, post_id: post_one.id, text: 'cuowevg owvwuei huiobqugog ey cfgiugey efuuioqcnfu qrb ryqfo hpe...')
+    Comment.create(user_id: user_two.id, post_id: post_two.id, text: 'cuowevg owvwuei huiobqugog ey cfgiugey efuuioqcnfu qrb ryqfo hpe...')
+end
```
Comment (BenMukebo) [inline db/seeds.rb:26]: ## [OPTIONAL]
- [ ] As I have said In the comment above kindly revise your association, here in the seed file you are using user_id to associate the post to the User but in your database, the association of the User and the Post is done using the author_id foreigner_key. Kindly revise this and do the same for the Comment

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @BANCUNGUYE66 

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 
- Professional README ✔️ 
- You have followed  JavaScript best practices. ✔️ 
- Descriptive README file.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb
Comment (franclobo) [inline app/models/post.rb:1]: - [ ] Well done completing the required changes by the last CR Kindly correct this method 👇🏽 

**Incorrect method**

```ruby
def update_posts_counter
  user.increment!(:posts_counter)
end
```
**Correct method** 
``` ruby
def update_posts_counter
 author.increment!(:posts_counter)
end
``` 

to correct this issue 👇🏽 


```ruby
rails aborted!
NameError: undefined local variable or method `user' for #<Post id: nil, author_id: 1, title: "Rerum eos amet.", text: "Ut voluptatem harum. Nihil pariatur quaerat. Volup...", comments_counter: 0, likes_counter: 0, created_at: "2023-07-27 17:55:37.529987000 +0000", updated_at: "2023-07-27 17:55:37.529987000 +0000">
/home/franclobo/microverse/Blog-app/app/models/post.rb:13:in `update_posts_counter'
/home/franclobo/microverse/Blog-app/db/seeds.rb:22:in `block in <main>'
/home/franclobo/microverse/Blog-app/db/seeds.rb:21:in `times'
/home/franclobo/microverse/Blog-app/db/seeds.rb:21:in `<main>'
Tasks: TOP => db:seed
(See full trace by running task with --trace)
```

File: app/views/users/index.html.erb
Comment (franclobo) [inline app/views/users/index.html.erb:1]: - **[OPTIONAL]** Kindly add the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

File: app/views/users/show.html.erb
Comment (franclobo) [inline app/views/users/show.html.erb:1]: - **[OPTIONAL]** Kindly add the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

File: app/views/posts/index.html.erb
Comment (franclobo) [inline app/views/posts/index.html.erb:1]: - **[OPTIONAL]** KIndly add the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image

File: app/views/posts/show.html.erb
Comment (franclobo) [inline app/views/posts/show.html.erb:1]: - [ ] Kindly add the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @BANCUNGUYE66, 

# Required Changes ♻️

istockphoto-1079018918-612x612

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Correct views ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb
Comment (Abenezer-Tilahun) [reply app/models/post.rb:1]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @BANCUNGUYE66 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Comment by iambenkis** [comment]
Hi @BANCUNGUYE66  , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 

- [ ] You have submitted a project that is already merged, Kindly open another Pull request and then ask for a review 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me `@iambenkis` in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by FrederickMih — COMMENTED** [review]
## Hi :raising_hand: :raising_hand: @BANCUNGUYE66 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 


## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**